v2/apiserver/internal/core/events.go

```diff
@@ -549,6 +563,27 @@ func (e *eventsService) CancelMany(
 	// efficient way to do this?
 	go func() {
 		for _, event := range events.Items {
+			// Only iterate through a worker's jobs if the worker status is not
+			// Pending; otherwise no jobs would have been created.
+			if event.Worker.Status.Phase != WorkerPhasePending {
```

Comment:
I'd need to double-check the store code-- you may have already. Are we sure the events we're iterating over here have the _pre_-cancellation phases?

Reply:
Oh great point... they may indeed be post-cancellation -- well, most assuredly since immediately prior to this go func, we issue the event cancellation 😂 

Let me look at what we can do.  May also try your suggestion of building complex db queries in the event cancellation functions to include job cancellations...

Reply:
Ok, as it stands, we don't have a way to discern the phases of the original events (designated by the provided selector).  We could do so if we added a GetMany() (or similar) method to the events store to first get them and inspect, so we have this info by the time we run the loop.  Not sure if these added calls would be worth the exchange for not running the job cancellation loop if an event phase was pending (pre-cancellation)...